lib/PreviewProvider.js

```diff
@@ -211,9 +169,10 @@ PreviewProvider.prototype = {
       this._asyncSaveLinks(processedLinks, event);
     }
 
-    let cachedLinks = this._getEnhancedLinks(processedLinks, previewsOnly, event);
-    return this._getFaviconColors(cachedLinks).then(linksToSend => {
-      return linksToSend;
+    return this._asyncGetEnhancedLinks(processedLinks, previewsOnly, event).then(cachedLinks => {
+      return this._getFaviconColors(cachedLinks).then(linksToSend => {
+        return linksToSend;
```

Comment:
we compute some colors based on the tippy top site that we we're using to display it nicely before we have embedly's colours. that way we don't have to use the grey fallback color